Renewable output [3/3]

flomnes opened the pull request:
Write renewable input data into the `output` directory 
* year by year
* synthesis
* record years

For each time granularity
* hourly
* daily
* weekly
* monthly
* annual

flomnes committed:
Rework simulation.cpp's mapping

flomnes committed:
Add renewable data structures

flomnes committed:
Defines / prepro

flomnes committed:
Enum -> string conversion function

flomnes committed:
Remove coupling

flomnes committed:
Fix cluster.cpp

flomnes committed:
Fix build

flomnes committed:
Remove prepro files

flomnes committed:
Format sources

flomnes committed:
Load renewable TS

This reverts commit 4ae02b22f83812d027353861cc2f0cabeb761950.

flomnes committed:
Merge remote-tracking branch 'origin/develop' into feature/renewable_internal_structures

flomnes committed:
Fix build

flomnes committed:
Add roof-top and "other" 2, 3 and 4

flomnes committed:
Remove debug message

flomnes committed:
Look for renewable data in input/renewables (not renewable)

flomnes committed:
Fix warnings related to renewable's area.ini

flomnes committed:
Fix container.hxx

flomnes committed:
Residual load: include renewable clusters

flomnes committed:
Fix TS alloc

flomnes committed:
Enable SC builder for renewable

flomnes committed:
Area: call renewable actions

flomnes committed:
Remove obsolete comments

flomnes committed:
Flush renewable TS (swap flavour)

flomnes committed:
Add TODO note

flomnes committed:
Remove unused code

Renewable clusters did not exist in versions < 390

flomnes committed:
Force loading of renewable TS

Since TS generation is not possible

flomnes committed:
Merge branch 'feature/renewable_internal_structures' into feature/renewable_scenario_builder

flomnes committed:
Rename State::cluster -> State::thermalCluster

flomnes committed:
Remove unused header

flomnes committed:
Introduce renewable cluster in State

flomnes committed:
Remove duplicate defs of hourForEachThermalCluster

All variable classes inherit from IVariable. Most define the same
hourForEachThermalCluster implementation. For these, just inherit the
base function.

Total number of duplicate definitions removed: 48

Script used:
SOURCE_DIRS="analyzer/ ext/ libs/ solver/ tools/ ui/"
SOURCE_FILES=$(find $SOURCE_DIRS -type f \( -name "*.h" -or -name "*.cpp" -or -name "*.h" -or -name "*.hxx" -or -name "*.hpp" -or -name "*.cxx" -or -name "*.c" \) -and -not \( -iname "sqlite3.c" -or -name "*.h.generator.hpp" -or -name "*.hxx.generator.hpp" -or -name "*.generator.commonstypes.hpp" \))
echo $SOURCE_FILES | xargs sed -i -z 's/    void hourForEachThermalCluster(State& state, unsigned int numSpace)\n    {\n        \/\/ Next item in the list\n        NextType::hourForEachThermalCluster(state, numSpace);\n    }\n//g'

flomnes committed:
De-duplicate function initializeFromThermalCluster

Number of occurences: 60

flomnes committed:
Remove duplicate yearEndBuildPrepareDataForEachThermalCluster

flomnes committed:
Add renewableGeneration.h

flomnes committed:
De-duplicate yearEndBuildForEachThermalCluster

flomnes committed:
Add (zero) output for renewables

flomnes committed:
Apply clang-format

flomnes committed:
Fill output data for renewable clusters

flomnes committed:
Fix previous commit

flomnes committed:
{Thermal,Renewable}ClusterName -> ClusterName

flomnes committed:
UI : Add timeSeriesRenewable to mapping

flomnes committed:
Remove disabled renewable clusters before solver is called

flomnes committed:
Remove log space

flomnes committed:
Remove RenewableClusterList::mapping

Redundant with RenewableClusterList::cluster

flomnes committed:
Remove switch...case in string -> groupID conversion

Clunky optimization

flomnes committed:
Write renewable cluster list and TS

flomnes committed:
Use a common data structure for renewable & thermal TS

flomnes committed:
Apply clang-format

flomnes committed:
Remove DataSeries{Thermal,Renewable}

flomnes committed:
Improve memory estimation for renewable TS

guilpier-code commented on src/solver/variable/area.inc.hxx:
```diff
@@ -241,6 +250,16 @@ void Areas<NEXTTYPE>::yearEndBuild(State& state, uint year, uint numSpace)
+        for (uint j = 0; j != area.renewable.clusterCount; ++j)
+        {
+            state.renewableCluster = area.renewable.clusters[j];
+            state.yearEndReset();
```
**yearEndReset** has already been called for thermal clusters. Calling it here again could set to zero some results computed for thermal clusters.

flomnes replied:
I duplicated this function into two : one for thermal data, one for renewable data. See commit 9573759.

guilpier-code commented on src/solver/variable/area.inc.hxx:
```diff
@@ -229,7 +238,7 @@ void Areas<NEXTTYPE>::yearEndBuild(State& state, uint year, uint numSpace)
+            state.thermalCluster = area.thermal.clusters[j];
             state.yearEndReset();
 
             // Variables
```
**yearEndBuildFromThermalClusterIndex** function name is very unclear on what the function really does.
This name should be changed, even if it has nothing to see with the current PR.

flomnes replied:
How about **yearEndRecoverThermalData** ?

guilpier-code commented on src/solver/variable/commons/links/links.hxx.inc.hxx:
```diff
@@ -104,6 +104,21 @@ inline void Links::yearEndBuildPrepareDataForEachThermalCluster(State& state,
     }
 }
 
+inline void Links::yearEndBuildPrepareDataForEachRenewableCluster(State& state,
```
This function is useless for now, but might be used someday (to be checked).

flomnes replied:
This function is used to recover hourly data for renewable clusters, see **src/solver/variable/economy/renewableGeneration.h:290**.
```cpp
void yearEndBuildPrepareDataForEachRenewableCluster(State& state,
                                                    uint year,
                                                    unsigned int numSpace)
{
    for (unsigned int i = 0; i <= state.study.runtime->rangeLimits.hour[Data::rangeEnd]; ++i)
    {
        state.renewableClusterProductionForYear[i]
          += pValuesForTheCurrentYear[numSpace][state.renewableCluster->areaWideIndex].hour[i];
    }
 // Next variable
    NextType::yearEndBuildPrepareDataForEachRenewableCluster(state, year, numSpace);
}
```

flomnes committed:
Fix header comparison

flomnes committed:
Merge branch 'feature/renewable_internal_structures' into feature/renewable_output

flomnes committed:
Add renewable test

flomnes committed:
Fix tests

flomnes committed:
Use pandas instead of numpy to compare outputs

It allows for more flexibility when adding new variables because after
this change, we fetch columns by name, not by index.

flomnes committed:
Remove useless try..except

flomnes committed:
Add missing helper trim_column_name.py

flomnes committed:
Hydro example tests : use "H. LEV" instead of index 26

flomnes committed:
Output comparison tests : print missing column names

flomnes committed:
Split State::yearEndReset into two functions

One for thermal data, one for renewable data.

flomnes commented:
Local merge in PR #267
